### PR TITLE
filtering and refactoring

### PR DESCRIPTION
**Pull request by Rocombol**

### CODE REVIEW

File: src/app/participants/participants.controller.js [hunk 1]
```diff
@@ -19,16 +19,17 @@ export class ParticipantsController {
...
+            size: 'lg',
+            resolve: {currentParticipant: () => participant}
         });
 
-        modalInstance.result.then((participant) => {
-         let id = currentParticipant.id;
-
-            this.http.put('/participants/' + id, participant)
+        modalInstance.result.then((currentParticipant) => {
+            var id = participant.id;
```
Comment (ksenya-ta) [inline src/app/participants/participants.controller.js]: why var?

File: src/app/participants/participants.html [hunk 2]
```diff
@@ -1,23 +1,42 @@
...
                 <td><button class="btn-danger" ng-click="participants.removePerson(participant.id)">Delete</button></td>
             </tr>
         </tbody>
     </table>
-
+    <label>Search Participant:
+    <input type="text" id="query" ng-model="query"/>
+    </label>
+    <label class="formgroup">Order by:
+      <select ng-model="participants.orderProp">
```
Comment (ksenya-ta) [inline src/app/participants/participants.html]: you can try using bootstrap's dropdown for this
http://getbootstrap.com/components/#dropdowns

Comment (Rocombol) [inline src/app/participants/participants.html]: Those types of bootstrap dropdown don`t work in our build :(

File: src/app/participants/UnicForm.html [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        <div class = "form-group">
+            <label class = "col-sm-2 control-label">Name</label>
+            <div class = "col-sm-10">
+                 <input type = "text"
+                     name = "name" 
+                     class = "form-control"
+                     ng-model = "participant.name"
+                     ng-maxlength = "15"
+                     placeholder= "Name"
+                     ng-required = "true">
```
Comment (ksenya-ta) [inline src/app/participants/UnicForm.html]: use just required(not ng-required) if it is always required(your case), ng-required is more for conditional validation

File: src/app/participants/participants.controller.js [hunk 4]
```diff
@@ -1,34 +1,39 @@
...
-  constructor($http,  $uibModal) {
+  constructor($http,  $uibModal, $scope) {
     'ngInject';
 
     this.http = $http;
-
     this.modal = $uibModal;
-
     this.getParticipants();
+    this.scope = $scope;
```
Comment (ksenya-ta) [inline src/app/participants/participants.controller.js]: why do you need $scope here?

File: src/app/participants/participants.html [hunk 5]
```diff
@@ -1,23 +1,43 @@
...
+            <tr class = "active" ng-repeat="participant in participants.participantsList | filter:query | orderBy: participants.orderProp track by participant.id">
+                <td>{{$index+1}}</td>
                 <td>{{participant.name}}</td>
                 <td>{{participant.lastname}}</td>
-                <td><button class="btn-info" ng-click="participants.editPerson(participant)">Edit</button></td>
-                <td><button class="btn-danger" ng-click="participants.removePerson(participant.id)">Delete</button></td>
+                <td>{{participant.company}}</td>
+                <td>{{participant.direction}}</td>
+                <td>{{participant.age}}</td>
+                <td><button class="btn-info" ng-click="participants.editPerson(participant)"><span class="glyphicon glyphicon-pencil"></span></button></td>
```
Comment (ksenya-ta) [inline src/app/participants/participants.html]: i tag for icons

File: src/app/participants/participants.html [hunk 6]
```diff
@@ -1,23 +1,43 @@
...
+                <td><button class="btn btn-sm btn-danger" ng-click="participants.removePerson(participant.id)"><span class="glyphicon glyphicon-trash"></span></button></td>
             </tr>
         </tbody>
     </table>
-
-<button class="btn btn-info" ng-click="participants.newPerson()">Add person</button>
+    <div   class="dropdown">
+        <div  class="btn-group" uib-dropdown is-open="status.isopen">
+          <button   id="single-button" type="button" class="btn btn-primary" uib-dropdown-toggle ng-disabled="disabled">
+            Button dropdown <span class="caret"></span>
```
Comment (ksenya-ta) [inline src/app/participants/participants.html]: button dropdown? rename it - it is not obvious what we need it for. If it was for sorting than it is better to place it above the table

File: src/app/participants/participants.controller.js [hunk 7]
```diff
@@ -1,34 +1,39 @@
...
+    this.newParticipant = {};
   }
 
+    applyFilter(filter){
```
Comment (ksenya-ta) [inline src/app/participants/participants.controller.js]: rename this method - it sets sorting property, not applying the filters

File: src/app/participants/UnicForm.html [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        <div class = "form-group">
+            <label class = "col-sm-2 control-label">Name</label>
+            <div class = "col-sm-10">
+                 <input type = "text"
+                     name = "name" 
+                     class = "form-control"
+                     ng-model = "participant.name"
+                     ng-maxlength = "15"
+                     placeholder= "Name"
+                     required = "true">
```
Comment (ksenya-ta) [inline src/app/participants/UnicForm.html]: just a required

**Pull request merged**